python/examples/jsonschema/pointcloud_csv_to_mcap.py

```diff
@@ -0,0 +1,118 @@
+""" Example script for converting pointcloud CSV data to an MCAP viewable in Foxglove Studio.
+
+This script uses the public CSV dataset "Sydney Urban Objects Dataset",
+released by the Australian Centre for Field Robotics at the University of Sydney, NSW, Australia.
+The original dataset can be downloaded from here:
+https://www.acfr.usyd.edu.au/papers/SydneyUrbanObjectsDataset.shtml
+
+usage:
+    curl https://www.acfr.usyd.edu.au/papers/data/sydney-urban-objects-dataset.tar.gz | tar -x
+    pip install mcap
+    python3 pointcloud_csv_to_mcap.py sydney-urban-objects-dataset/objects/4wd.0.2299.csv -o 4wd.mcap
+"""
+import argparse
+import base64
+import csv
+import json
+import typing
+import struct
+from pathlib import Path
+import datetime
+
+# tutorial-mcap-imports-start
+from mcap.mcap0.writer import Writer
+from mcap.mcap0.well_known import SchemaEncoding, MessageEncoding
+
+# tutorial-mcap-imports-end
+
+# tutorial-csv-decode-start
+def point_reader(csv_path: typing.Union[str, Path]):
+    with open(csv_path, "r") as f:
+        for timestring, i, _, x, y, z, _, _, _ in csv.reader(f):
+            timestamp = datetime.datetime.strptime(timestring, "%Y%m%dT%H%M%S.%f")
+            yield (timestamp, float(i), float(x), float(y), float(z))
+            # tutorial-csv-decode-end
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("csv", help="The input CSV to read")
+    parser.add_argument(
+        "--output", "-o", default="out.mcap", help="The MCAP output path to write"
+    )
+    args = parser.parse_args()
+
+    pointcloud: typing.Dict[str, typing.Any]
+    # tutorial-point-layout-start
+    float32 = 7  # as defined in the schema
+    pointcloud = {
+        "point_stride": (4 + 4 + 4 + 4),  # four bytes per float
+        "fields": [
+            {"name": "x", "offset": 0, "type": float32},
+            {"name": "y", "offset": 4, "type": float32},
+            {"name": "z", "offset": 8, "type": float32},
+            {"name": "i", "offset": 12, "type": float32},
+        ],
+    }
+    # tutorial-point-layout-end
+
+    # tutorial-pack-points-start
+    points = bytearray([])
+    base_timestamp = None
+    for point_timestamp, intensity, x, y, z in point_reader(args.csv):
+        if base_timestamp is None:
+            base_timestamp = point_timestamp
+        points.extend(struct.pack("<ffff", x, y, z, intensity))
```

Comment:
maybe reserve buffer space up front and use `pack_into`?

Reply:
I prefer this just because fewer lines of code in a tutorial setting is good.